app/assets/javascripts/discourse/controllers/preferences/interface.js.es6

```diff
@@ -109,6 +120,11 @@ export default Ember.Controller.extend(PreferencesTabController, {
         this.set("model.user_option.theme_ids", [this.get("themeId")]);
       }
 
+      const makeTextSizeDefault = this.get("makeTextSizeDefault");
+      if (makeTextSizeDefault) {
+        this.set("model.user_option.text_size", this.get("textSize"));
```

Comment:
I don’t know how @eviltrout feels about this but I try more and more to avoid using "set" inside a computed property. It makes code harder to read and might lead to weird bugs; eg: you have to make sure you are not using  `model.user_option.text_size` for something which is used to define the property you are in, which in some cases might be in other functions.

In this case can’t we use some init hook?

Reply:
Agreed, I blindly followed the existing convention for the theme setting here. I will see about moving both to an init hook.